rds/rds_provider.go

```diff
@@ -21,22 +21,36 @@ type RDS struct {
 // CreateDatabase creates a database from the CRD database object, is also ensures that the correct
 // subnets are created for the database so we can access it
 func (r *RDS) CreateDatabase(db *crd.Database, password string) (string, error) {
-	// Ensure that the subnets for the DB is create or updated
-	log.Println("Trying to find the correct subnets")
-	subnetName, err := r.ensureSubnets(db)
-	if err != nil {
-		return "", err
+	if db.Spec.DBSnapshotIdentifier == "" {
+		return r.CreateDatabaseInstance(db, password)
+	} else {
+		return r.RestoreDatabaseFromSnapshot(db, password)
```

Comment:
I guess there needs to be a check on the status section of the spec too. I case someone updates a spec with the `DBSnapshotIdentifier` already set.
So something like:
`if DBSnapshotIdentifier == "" and db.Status != "Created"`

Reply:
I see your point but adding `db.Status != "Created"` will also prevent the `CreateDatabaseInstance` to be run if already created (idempotence seems to already be managed by `CreateDatabaseInstance` for creation at AWS level). But it would also allow to run `RestoreDatabaseFromSnapshot` if `DBSnapshotIdentifier` is empty string if db is created.
We may add a spec or an annotation (or even resource status) to specify it was created from a restore.

Idempotence and behavior in such update case is more or less related to error handling imho. Don't know when and where we should catch and react to them. Would be nice to talk about it :)